scripts/zkai_attention_kv_stwo_typed_size_estimate_gate.py

```
@@ -0,0 +1,684 @@
+        )
+    try:
+        return json.loads(completed.stdout)
+    except json.JSONDecodeError as err:
+        raise StwoTypedSizeEstimateGateError(f"Stwo size estimate CLI did not emit JSON: {err}") from err
+
+
+def build_rows() -> list[dict[str, Any]]:
+    specs = artifact_specs(section_delta_rows())
+    cli = run_stwo_size_estimate([spec["path"] for spec in specs])
+    if cli.get("schema") != "zkai-stwo-proof-size-estimate-cli-v1":
+        raise StwoTypedSizeEstimateGateError("CLI schema drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)